src/Validation/SettingsCaptcha.php

```diff
@@ -0,0 +1,227 @@
+<?php
+
+/**
+ * Captcha Settings class - Google reCaptcha.
+ *
+ * @package EightshiftForms\Validation
+ */
+
+declare(strict_types=1);
+
+namespace EightshiftForms\Validation;
+
+use EightshiftForms\Hooks\Variables;
+use EightshiftForms\Settings\SettingsHelper;
+use EightshiftForms\Labels\LabelsInterface;
+use EightshiftForms\Settings\Settings\SettingsDataInterface;
+use EightshiftFormsVendor\EightshiftLibs\Services\ServiceInterface;
+
+/**
+ * SettingsCaptcha class.
+ */
+class SettingsCaptcha implements SettingsDataInterface, ServiceInterface
+{
+	/**
+	 * Use general helper trait.
+	 */
+	use SettingsHelper;
+
+	/**
+	 * Filter settings sidebar key.
+	 */
+	public const FILTER_SETTINGS_SIDEBAR_NAME = 'es_forms_settings_sidebar_captcha';
+
+	/**
+	 * Filter global settings key.
+	 */
+	public const FILTER_SETTINGS_GLOBAL_NAME = 'es_forms_settings_global_captcha';
+
+	/**
+	 * Filter settings global is Valid key.
+	 */
+	public const FILTER_SETTINGS_GLOBAL_IS_VALID_NAME = 'es_forms_settings_global_is_valid_captcha';
+
+	/**
+	 * Settings key.
+	 */
+	public const SETTINGS_TYPE_KEY = 'captcha';
+
+	/**
+	 * Captcha Use key.
+	 */
+	public const SETTINGS_CAPTCHA_USE_KEY = 'captcha-use';
+
+	/**
+	 * Google reCaptcha site key.
+	 */
+	public const SETTINGS_CAPTCHA_SITE_KEY = 'captcha-site-key';
+
+	/**
+	 * Google reCaptcha secret key.
+	 */
+	public const SETTINGS_CAPTCHA_SECRET_KEY = 'captcha-secret-key';
+
+	/**
+	 * Google reCaptcha score key.
+	 */
+	public const SETTINGS_CAPTCHA_SCORE_KEY = 'captcha-score';
+
+	/**
+	 * Google reCaptcha score default key.
+	 */
+	public const SETTINGS_CAPTCHA_SCORE_DEFAULT_KEY = 0.5;
+
+	/**
+	 * Instance variable for labels data.
+	 *
+	 * @var LabelsInterface
+	 */
+	protected $labels;
+
+	/**
+	 * Create a new instance.
+	 *
+	 * @param LabelsInterface $labels Inject documentsData which holds labels data.
+	 */
+	public function __construct(LabelsInterface $labels)
+	{
+		$this->labels = $labels;
+	}
+
+	/**
+	 * Register all the hooks
+	 *
+	 * @return void
+	 */
+	public function register(): void
+	{
+		\add_filter(self::FILTER_SETTINGS_SIDEBAR_NAME, [$this, 'getSettingsSidebar']);
+		\add_filter(self::FILTER_SETTINGS_GLOBAL_NAME, [$this, 'getSettingsGlobalData']);
+		\add_filter(self::FILTER_SETTINGS_GLOBAL_IS_VALID_NAME, [$this, 'isSettingsGlobalValid']);
+	}
+
+	/**
+	 * Determine if settings global are valid.
+	 *
+	 * @return boolean
+	 */
+	public function isSettingsGlobalValid(): bool
+	{
+		$isUsed = (bool) $this->isCheckboxOptionChecked(self::SETTINGS_CAPTCHA_USE_KEY, self::SETTINGS_CAPTCHA_USE_KEY);
```

Comment:
Just a thought: could `isCheckboxOptionChecked` method return `boolean` value by default? In that case, you wouldn't need to manually cast it to a boolean 🤷🏼‍♂️

Reply:
it is returning bool with type hint but for some reason stan is yelling at me :(

Reply:
It looks like it is fixed. I have updated this on multiple instance